app/controllers/Checkout.scala

```diff
@@ -45,35 +45,24 @@ object Checkout extends Controller with LazyLogging {
     SubscriptionsForm().bindFromRequest.fold(
       formWithErrors => {
         for (idUserOpt <- getIdentityUserByCookie(request))
-          yield BadRequest(views.html.checkout.payment(formWithErrors, userIsSignedIn = idUserOpt.isDefined))
+          yield BadRequest(view.payment(formWithErrors, userIsSignedIn = idUserOpt.isDefined))
       },
       userData => {
         val idUserOpt = AuthenticationService.authenticatedUserFor(request)
-        CheckoutService.processSubscription(userData, idUserOpt).map {
-          case CheckoutResult(_, guestUser: GuestUser, _) =>
-            Some(FinishAccountForm().bind(guestUser.toFormParams))
-          case CheckoutResult(_, _:MinimalIdUser, _) =>
-            None
-        }.map(formOpt => Ok(views.html.checkout.thankyou(formOpt)))
+        CheckoutService.processSubscription(userData, idUserOpt).map { case CheckoutResult(_, userIdData, subscription) =>
+          val passwordForm = userIdData.toGuestAccountForm
+          Ok(view.thankyou(subscription.id, userData.personalData, passwordForm))
+        }
       }
     )
   }
 
-  def thankyou = GoogleAuthenticatedStaffAction.async { implicit request =>
-    Future {
-      val form = FinishAccountForm().bindFromRequest()
-      Ok(views.html.checkout.thankyou(Some(form)))
-    }
-  }
-
   def processFinishAccount = GoogleAuthenticatedStaffAction.async { implicit request =>
-    FinishAccountForm().bindFromRequest.fold( formWithErrors => {
-      Future {
-        BadRequest(views.html.checkout.thankyou(Some(formWithErrors)))
-      }
+    FinishAccountForm().bindFromRequest.fold({ _ =>
+      throw new RuntimeException("Could not set a password for guest account. Validation should happen on the client side!")
     }, guestAccountData => {
       IdentityService.convertGuest(guestAccountData.password, IdentityToken(guestAccountData.token))
-        .map(_ => Ok(views.html.checkout.alldone()))
+        .map(_ => Ok(view.alldone()))
```

Comment:
Yeah, that is due to some ambiguity in the visual specification of the guest user flow (I had referred to wireframes in Digital Pack v2, not to the more recent clickable wireframes). @davidrapson has agreed to pick up this bit and to replace the All done page with an Ajax endpoint.